Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py

```diff
@@ -0,0 +1,2284 @@
+            method='GET',
+            url_suffix='/api/data/threat_intelligence/IOCIndicator/',
+            params=data
+        )
+
+    def add_ioc_to_source(self, ioc_value, ioc_type, ioc_comment, ioc_status, source_id):
+
+        testing_status = None
+
+        if ioc_status == 'testing':
+            testing_status = 'in_progress'
+
+        data = {
+            'type': ioc_type,
+            'value': ioc_value,
+            'comment': ioc_comment,
+            'source_id': source_id,
+            'hl_status': ioc_status,
+            'hl_local_testing_status': testing_status
+        }
+
+        return self._http_request(
+            method='POST',
+            url_suffix='/api/data/threat_intelligence/IOCIndicator/',
+            json_data=data
+        )
+
+    def delete_ioc(self, ioc_id):
+        return self._http_request(
+            method='DELETE',
+            url_suffix=f'/api/data/threat_intelligence/IOCIndicator/{ioc_id}/',
+            return_empty_response=True
+        )
+
+    def assign_policy_to_agent(self, policyid, agentid):
+        data = {
+            'agent_ids': [agentid]
+        }
+
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/api/data/endpoint/Policy/{policyid}/add_agents/',
+            json_data=data
+        )
+
+
+def assign_policy_to_agent(client, args):
+
+    policy_name = args.get('policy', None)
+
+    results = client.list_policies(policy_name)
+    policyid = None
+    for policy in results['results']:
+        if args['policy'] == policy['name']:
+            policyid = policy['id']
+            break
+    if policyid:
+        client.assign_policy_to_agent(policyid, args['agentid'])
+
+
+def test_module(client, args):
+    result = client.test_api()
+    if 'version' in result:
+        demisto.results('ok')
+    else:
+        demisto.results('failed to access version endpoint')
+
+
+def fetch_incidents(client, args):
+
+    last_run = demisto.getLastRun()
+
+    if 'first_fetch' in args and args['first_fetch']:
+        days = int(args['first_fetch'])
+    else:
+        days = 0
+    first_fetch_time = int(datetime.timestamp(datetime.now() - timedelta(days=days)) * 1000000)
+    alert_status = args.get('alert_status', None)
+    alert_type = args.get('alert_type', None)
+    min_severity = args.get('min_severity', SEVERITIES[0])
+    max_results = args.get('max_results', None)
+
+    severity = ','.join(SEVERITIES[SEVERITIES.index(min_severity):]).lower()
+
+    already_fetched_previous = []
+    already_fetched_current = []
+
+    last_fetch = None
+    if last_run:
+        last_fetch = last_run.get('last_fetch', None)
+        already_fetched_previous = last_run.get('already_fetched', [])
+
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch_time
+        last_fetch = first_fetch_time
+    else:
+        # otherwise use the stored last fetch
+        last_fetch = int(last_fetch)
```

Comment:
```suggestion
    if last_fetch is None:
        # if missing, use what provided via first_fetch_time
        last_fetch = first_fetch_time
    else:
        # otherwise use the stored last fetch
        last_fetch = min(int(last_fetch), first_fetch_time)
```
Add a check that you take the minimum between the two values.
It is needed for example, when the user may change the `first_fetch` in the integration's settings to be 1 day from 5 days, when the `last_fetch` is more than 1 day. In that case, we will want to take the minimum between the 2- meaning taking the `first_fetch`.

Reply:
Not clear whether we need to take the min or max. However, this parameter is just used for the first run of the connector. Then it is never taken into account. 
I prefer not to change it for the moment.